cluster/gce/gci/configure-helper.sh

```diff
@@ -66,33 +66,133 @@ function create-dirs {
+
+      if [[ ${opformat,,} == ${format,,} && ${opinterface,,} == ${interface,,} ]]; then
+        total=$((total+opnum))
+      fi
+    done
+  fi
+  
+}
+
+function safe-block-symlink(){
+  device=$1
+  interface=$2
+  mkdir -p "/mnt/disks/by-uuid/google-local-ssds-${interface,,}-block"
+  ssdmap="/home/kubernetes/localssdmap.txt"
+  echo "Creating symlink for block devices..."
+
+  if [[ ! -e ${ssdmap} ]]; then
+    touch ${ssdmap}
+    chmod +w ${ssdmap}
+  fi
+
+  # each line looks like "${device} persistant-uuid"
+  if [[ ! -z $(cat ${ssdmap} | grep ${device}) ]]; then
+    #create symlink based on saved uuid
+    myuuid=$(cat ${ssdmap} | grep ${device} | cut -d ' ' -f 2)
+  else
+    # generate new uuid and add it to the map
+    myuuid=$(uuidgen)
+    echo "${device} ${myuuid}" >> ${ssdmap}
+  fi
+
+  sym="/mnt/disks/by-uuid/google-local-ssds-${interface,,}-block/local-ssd-${myuuid}"
+  # dont make the exact directory first or else symlink goes 1 directory down
+  ln -s ${device} ${sym}
+  echo "Created a symlink for SSD $ssd at ${sym}"
+  chmod a+w ${sym}
+}
+
+
 # Formats the given device ($1) if needed and mounts it at given mount point
 # ($2).
-function safe-format-and-mount() {
+function safe-format-and-mount-by-interface() {
   device=$1
-  mountpoint=$2
+  interface=$2
 
   # Format only if the disk is not already formatted.
   if ! tune2fs -l "${device}" ; then
     echo "Formatting '${device}'"
     mkfs.ext4 -F "${device}"
   fi
 
+  # Trigger udev refresh so that newly formatted devices are propagated in by-uuid
+  udevadm control --reload-rules
+  udevadm trigger
+  udevadm settle
+  
+  # Get the real UUID of the device
+  item=$(readlink -f ${device} | cut -d '/' -f 3)
+  myuuid=$(ls -l /dev/disk/by-uuid/ | grep ${item} | tr -s ' ' | cut -d ' ' -f 9) 
+  mountpoint="/mnt/disks/by-uuid/google-local-ssds-${interface,,}-fs/local-ssd-${myuuid}"
+
+  # Mount device to the mountpoint
   mkdir -p "${mountpoint}"
   echo "Mounting '${device}' at '${mountpoint}'"
   mount -o discard,defaults "${device}" "${mountpoint}"
+  chmod a+w ${mountpoint}
+
+  # Set up a bind-mount in /mnt/disks/ for backwards compatability for all filesystem formatted SSDs
+  if [[ ${interface,,} == "scsi" ]]; then
+    scsinum=`echo ${device} | sed -e 's/\/dev\/disk\/by-id\/google-local-ssd-\([0-9]*\)/\1/'`
+    bindpoint="/mnt/disks/ssd${scsinum}"
+    echo "Bind mounting local SSD $mountpoint to $bindpoint"
+    mkdir -p ${bindpoint}
+    mount --bind  ${mountpoint} ${bindpoint}
+  else
+    nvmenum=`echo ${device} | sed -e 's/\/dev\/disk\/by-id\/google-nvme0n\([0-9]*\)/\1/'`
+    bindpoint="/mnt/disks/ssd-nvme${nvmenum}"
+    echo "Bind mounting local SSD $mountpoint to ${bindpoint}"
+    mkdir -p ${bindpoint}
+    mount --bind  ${mountpoint} ${bindpoint}
+  fi
+  
+}
+
+function mount-ext(){
+  ssd=$1
+  interface=$2
+  format=$3  
+
+  if [[ ${format,,} == "fs" ]]; then
+    safe-format-and-mount-by-interface ${ssd} ${interface,,}
```

Comment:
What if we call safe-format-and-mount like before, and then a new function that does the bind mount to the uuid path?

Reply:
this method is more natural for if we plan to eventually deprecate the "old" mount points. We would just have to delete the bind-mounting section as opposed to reversing the logic when the backwards compatibility is removed.

Reply:
I don't think we'll be able to deprecate it anytime soon.  We'll have to support both types of users.  Those that want to use hostpath and needs the paths to be consistent across nodes, and those that want Local PVs which needs all the paths to be unique.